src/control/controlobjectscript.cpp

```diff
@@ -127,13 +127,27 @@ void ControlObjectScript::disconnectAllConnectionsToFunction(const QJSValue& fun
     }
 }
 
+bool ControlObjectScript::isCallbackExecuting() const {
+    return m_callbackExecuting;
+}
+
+void ControlObjectScript::setCallbackExecuting(bool executing) {
+    m_callbackExecuting = executing;
+}
+
 void ControlObjectScript::slotValueChanged(double value, QObject*) {
     // Make a local copy of m_connectedScriptFunctions first.
     // This allows a script to disconnect a callback from inside the
-    // the callback. Otherwise the this may crash since the disconnect call
-    // happens during conn.function.call() in the middle of the loop below.
-    const QVector<ScriptConnection> connections = m_scriptConnections;
+     const QVector<ScriptConnection> connections = m_scriptConnections;
+    // Flag to inhibit triggering connections during callback execution
+
     for (auto&& conn: connections) {
-        conn.executeCallback(value);
+        if (!this->m_callbackExecuting) {
+            this->m_callbackExecuting = true;
+            conn.executeCallback(value);
+            this->m_callbackExecuting = false;
+        } else {
+            qCWarning(m_logger) << "Warning: trigger() called inside the callback.";
```

Comment:
This is a mapping script error and shall not happen in a release version of the script. 
Can we make this a critical error and rephrase the string accordingly?

Reply:
Could I use std::cerr for which I have to include iostream??

Reply:
No, just qCCritical instead of qCWarning.

Reply:
The error should be reported, in the trigger function - not here.

Reply:
Do you mean in triggerScriptConnection()? 

I am afraid we want two things with a single flag. Currently the flag is a ControlObjectScript member guarding recursive valueChange calls. If you want to guard recursive connection calls, we need to move the flag to ControllerScriptInterfaceLegacy.